plugins/modules/ntnx_clusters.py

```diff
@@ -0,0 +1,522 @@
+                "kind": "cluster",
+                "last_update_time": "2022-06-02T19:15:48Z",
+                "owner_reference": {
+                    "kind": "user",
+                    "name": "admin",
+                    "uuid": "00000000-0000-0000-0000-000000000000"
+                },
+                "spec_version": 51,
+                "uuid": "0005dec1-a7d4-a09e-3740-2cea7f9c9e08"
+            },
+            "spec": {
+                "name": "cluster_name",
+                "resources": {
+                    "config": {
+                        "authorized_public_key_list": []
+                    }
+                }
+            }
+        }
+"""
+
+from ..module_utils.base_module import BaseModule  # noqa: E402
+from ..module_utils.prism.clusters import Cluster  # noqa: E402
+from ..module_utils.prism.tasks import Task  # noqa: E402
+from ..module_utils.utils import remove_param_with_none_value  # noqa: E402
+
+
+def get_module_spec():
+    mutually_exclusive = [("name", "uuid")]
+
+    entity_by_spec = dict(name=dict(type="str"), uuid=dict(type="str"))
+
+    domain_server_spec = dict(
+        name=dict(type="str"),
+        nameserver=dict(type="str"),
+        domain_credentials=dict(
+            type="dict", options=dict(
+                username=dict(type="str"),
+                password=dict(type="str", no_log=True),
+            )
+        )
+    )
+
+    http_proxy_list_spec = dict(
+        name=dict(type="str"),
+        address=dict(
+            type="dict", options=dict(
+                ip=dict(type="str"),
+                ipv6=dict(type="str"),
+                fqdn=dict(type="str"),
+                port=dict(type="int"),
+                is_backup=dict(type="bool"),
+            )
+        ),
+        credentials=dict(
+            type="dict", options=dict(
+                username=dict(type="str"),
+                password=dict(type="str", no_log=True),
+            ),
+        ),
+        proxy_type_list=dict(
+            type="list", elements="str",
+        )
+    )
+
+    smtp_server_spec = dict(
+        email_address=dict(type="str"),
+        server=dict(
+            type="dict", options=dict(
+                name=dict(type="str"),
+                address=dict(
+                    type="dict", options=dict(
+                        ip=dict(type="str"),
+                        ipv6=dict(type="str"),
+                        fqdn=dict(type="str"),
+                        port=dict(type="int"),
+                        is_backup=dict(type="bool"),
+                    ),
+                ),
+                credentials=dict(
+                    type="dict", options=dict(
+                        username=dict(type="str"),
+                        password=dict(type="str", no_log=True),
+                    ),
+                ),
+                proxy_type_list=dict(type="list", elements="str"),
+            )
+        ),
+        server_type=dict(type="str"),
```

Comment:
If this field only allows few values then we can mention choices, for ex. choices=["PLAIN", "SMTP"]